angular/src/lib/modal/modal.component.ts

```diff
@@ -25,108 +23,90 @@ export type SizeType =
-  @Output() whenHide = new EventEmitter();
-
-  @ViewChild('overlayGlobalTemplate') modalContentTemplate: TemplateRef<any>;
-
-  private overlayRef: OverlayRef;
-  private tp: TemplatePortal;
-
-  constructor(
-    private overlay: Overlay,
-    public viewContainerRef: ViewContainerRef,
-    private modalService: ModalService
-  ) { }
-
-  get classes() {
-    return {
-      'md-modal': true,
-      ['md-modal--' + this.sizeType]: true,
-      overlayerFix: this.sizeType === 'full',
-      [this.class]: this.class,
-    };
+  @Input()
+  set class(value) {
+    this._classList = value;
+  }
+  get class() {
+    return this._classList;
   }
 
-  ngOnInit() {
-    const strategy = this.overlay
-      .position()
-      .global()
-      .centerHorizontally()
-      .centerVertically();
-    const config = new OverlayConfig({
-      hasBackdrop: this.backdrop,
-      positionStrategy: strategy,
-    });
-    this.overlayRef = this.overlay.create(config);
-    this.tp = new TemplatePortal(
-      this.modalContentTemplate,
-      this.viewContainerRef
-    );
-
-    if (this.backdropClickExit) {
-      this.overlayRef.backdropClick().subscribe(() => {
-        this.dismissContent();
-      });
+  private _show: Boolean = false;
+  /** @option Show/hide modal | 'false' */
+  @Input()
+  set show(value: Boolean) {
+    this._show = value;
+    if ( this._show ) {
+      this._trapFocus();
     }
-
-    this.modalService.isModalOpened$.subscribe(isOpen => {
-      if (isOpen) {
-        this.showContent();
-      } else {
-        this.dismissContent();
-      }
-    });
-    this.modalService.setModalStatus(this.show);
   }
-
-  public showModal = () => {
-    this.modalService.setModalStatus(true);
+  get show() {
+    return this._show;
   }
 
-  public closeModal = () => {
-    this.modalService.setModalStatus(false);
-  }
+  /** @option Size of the modal | 'default' */
+  @Input() public sizeType: SizeType = 'default';
 
-  private showContent = () => {
-    if (this.overlayRef && !this.overlayRef.hasAttached()) {
-      this.overlayRef.attach(this.tp);
-    }
-  }
+  /** @option htmlId for modal | 'md-modal' */
+  @Input() public htmlId: String = 'md-modal';
+
+   /** @option ariaLabel for modal | '' */
+   @Input() public ariaLabel: String = '';
+
+   @Output() closeOnBackdrop: EventEmitter<any> = new EventEmitter<any>();
+
+   private _focusTrap: FocusTrap;
 
-  private dismissContent = () => {
-    if (this.overlayRef && this.overlayRef.hasAttached()) {
-      this.overlayRef.detach();
-      if (this.overlayRef.backdropElement) {
-        this.overlayRef.backdropElement.remove();
-      }
-      this.whenHide.emit();
+   constructor(
+     private _elRef: ElementRef,
+     private _focusTrapFactory: FocusTrapFactory,
+     @Optional() @Inject(DOCUMENT) private _document: any
+     ) {}
+
+   private _trapFocus() {
+     const element = this._elRef.nativeElement;
+
+     if ( !this._focusTrap ) {
```

Comment:
not a biggie but, there are some if statements in this file with spacing and some without. Best to keep it consistent

if (!this._focusTrap)